test/e2e/testsuites/testsuites.go

```diff
@@ -73,83 +72,14 @@ func (t *TestStorageClass) Create() storagev1.StorageClass {
 	var err error
 
 	ginkgo.By("creating a StorageClass " + t.storageClass.Name)
-	t.storageClass, err = t.client.StorageV1().StorageClasses().Create(t.storageClass)
+	t.storageClass, err = t.client.StorageV1().StorageClasses().Create(context.TODO(), t.storageClass, metav1.CreateOptions{})
 	framework.ExpectNoError(err)
 	return *t.storageClass
 }
 
 func (t *TestStorageClass) Cleanup() {
 	e2elog.Logf("deleting StorageClass %s", t.storageClass.Name)
-	err := t.client.StorageV1().StorageClasses().Delete(t.storageClass.Name, nil)
-	framework.ExpectNoError(err)
-}
-
-type TestVolumeSnapshotClass struct {
-	client              restclientset.Interface
-	volumeSnapshotClass *v1alpha1.VolumeSnapshotClass
-	namespace           *v1.Namespace
-}
-
-func NewTestVolumeSnapshotClass(c restclientset.Interface, ns *v1.Namespace, vsc *v1alpha1.VolumeSnapshotClass) *TestVolumeSnapshotClass {
-	return &TestVolumeSnapshotClass{
-		client:              c,
-		volumeSnapshotClass: vsc,
-		namespace:           ns,
-	}
-}
-
-func (t *TestVolumeSnapshotClass) Create() {
-	ginkgo.By("creating a VolumeSnapshotClass")
-	var err error
-	t.volumeSnapshotClass, err = snapshotclientset.New(t.client).VolumesnapshotV1alpha1().VolumeSnapshotClasses().Create(t.volumeSnapshotClass)
-	framework.ExpectNoError(err)
-}
-
-func (t *TestVolumeSnapshotClass) CreateSnapshot(pvc *v1.PersistentVolumeClaim) *v1alpha1.VolumeSnapshot {
```

Comment:
where is Snapshot case?

Reply:
The latest`external-snapshotter` just supports client-go v0.17. In client-go v0.18.0+, some interfaces add the parameter `context.Context`.  external-snapshotter will break.

But legacy-cloud-provider needs v0.18.0+.

Since there is not any snapshot related e2e test, delete the unnecessary codes now.